bin/improver-spotdata

```diff
@@ -0,0 +1,132 @@
+                        help='A list of diagnostics that are to be processed. '
+                             'If unset, all diagnostics defined in the '
+                             'config_file will be produced; e.g. '
+                             'temperature wind_speed')
+
+    # File based site definitions.
+    parser.add_argument('--site_path', type=str,
+                        help='Path to site data file if this is being used '
+                             'to choose sites.')
+    parser.add_argument('--constants_path', type=str,
+                        help='Path to json file containing constants to use '
+                             'in SpotData methods.')
+
+    # Run time site definition options.
+    parser.add_argument('--latitudes', type=valid_latitude, metavar='(-90,90)',
+                        nargs='+',
+                        help='List of latitudes of sites of interest.')
+    parser.add_argument('--longitudes', type=valid_longitude,
+                        metavar='(-180,180)', nargs='+',
+                        help='List of longitudes of sites of interest.')
+    parser.add_argument('--altitudes', type=float, nargs='+',
+                        help='List of altitudes of sites of interest.')
+    parser.add_argument('--site_ids', type=int, nargs='+',
+                        help='ID numbers for sites can be set if desired.')
+
+    # Forecast date/time information.
+    parser.add_argument('--forecast_date', type=str,
+                        help='Start date of forecast in format YYYYMMDD '
+                             '(e.g. 20170327 = 27th March 2017).')
+    parser.add_argument('--forecast_time', type=int,
+                        help='Starting hour of forecast in 24hr clock. '
+                             '(e.g. 3 = 03Z, 14 = 14Z).')
+    parser.add_argument('--forecast_length', type=int,
+                        help='Length of forecast in hours.')
+    parser.add_argument('--output_path', type=str,
+                        help='Path to which output files should be written.')
+    parser.add_argument('--multiprocess', type=bool,
+                        help='Process diagnostics using multiprocessing.')
+
+    args = parser.parse_args()
+
+    site_properties = {
+        'latitudes': args.latitudes,
+        'longitudes': args.longitudes,
+        'altitudes': args.altitudes,
+        'site_ids': args.site_ids
+        }
+
+    run_framework(args.config_file_path, args.data_path, args.ancillary_path,
+                  diagnostic_list=args.diagnostics, site_path=args.site_path,
+                  constants_file_path=args.constants_path,
+                  site_properties=site_properties,
+                  forecast_date=args.forecast_date,
+                  forecast_time=args.forecast_time,
+                  forecast_length=args.forecast_length,
+                  output_path=args.output_path,
+                  use_multiprocessing=args.multiprocess)
+
```

Comment:
Ultimately I think that 'framework.py' (now main.py) should be divided between the CLI and the suite, but the suite doesn't really exist in a fully functional form yet.